crates/miden-lib/asm/kernels/transaction/lib/account_delta.masm

```diff
@@ -99,6 +101,171 @@ export.compute_commitment
+
+        exec.constants::get_storage_slot_type_map eq
+        # => [is_map_slot_type, slot_ptr, slot_idx, init_slots_ptr]
+
+        if.true
+            dup.1 swap
+            # => [slot_ptr, slot_idx, slot_idx, init_slots_ptr]
+
+            # load the map root
+            padw movup.4 mem_loadw
+            # => [MAP_ROOT, slot_idx, slot_idx, init_slots_ptr]
+
+            # add map entries to the in-kernel delta
+            exec.add_storage_map_entries_to_delta
+            # => [slot_idx, init_slots_ptr]
+        else
+            drop
+            # => [slot_idx, init_slots_ptr]
+        end
+        # => [slot_idx, init_slots_ptr]
+
+        dup neq.0
+        # => [should_continue, slot_idx, init_slots_ptr]
+    end
+    # => [slot_idx, init_slots_ptr]
+
+    drop drop
+    # => []
+end
+
+#! Inserts the entries of the provided storage map root into the account delta.
+#!
+#! These entries must be present in the advice provider with the map root as the key.
+#! For each entry, the initial value in the delta is set to the empty word so the delta for this
+#! map is computed as if the map had initially been empty.
+#!
+#! The procedure proves inclusion of the map entries in the account's storage map at the provided
+#! index. Note that the advice map entry could contain multiple occurrences of a key-value pair
+#! A -> B, but this procedure has no way of detecting this condition.
+#! Such entries would naturally be deduplicated by the nature of the storage map or link map
+#! (used to track the storage map delta), and so are not problematic for overall correctness.
+#!
+#! Inputs:
+#!   Operand stack: [MAP_ROOT, slot_idx]
+#!   Advice map: { MAP_ROOT: [MAP_ENTRIES] }
+#! Outputs: []
+proc.add_storage_map_entries_to_delta
+    adv.push_mapvaln
+    # OS => [MAP_ROOT, slot_idx]
+    # AS => [num_entries, [MAP_ENTRIES]]
+
+    movup.4
+    # OS => [slot_idx, MAP_ROOT]
+    # AS => [num_entries, [MAP_ENTRIES]]
+
+    adv_push.1
+    # OS => [num_entries, slot_idx, MAP_ROOT]
+    # AS => [[MAP_ENTRIES]]
+
+    dup push.8 u32assert2.err="number of storage map entries should fit into a u32"
+    # OS => [8, num_entries, num_entries, slot_idx, MAP_ROOT]
+    # AS => [[MAP_ENTRIES]]
+
+    # check that num_entries % 8 = 0 so we can use an equality check for the loop condition
+    u32divmod eq.0 assert.err="number of storage map entries must be a multiple of 8"
+    drop
+    # OS => [num_entries, slot_idx, MAP_ROOT]
+    # AS => [[MAP_ENTRIES]]
+
+    # loop if there are more than 0 storage map entries
+    dup neq.0
+    # OS => [should_loop, num_entries, slot_idx, MAP_ROOT]
+    # AS => [[MAP_ENTRIES]]
+
+    while.true
+        # we process two words per iteration, so decrement by 8 felts
+        sub.8
+        # => [remaining_entries, slot_idx, MAP_ROOT]
+
+        # push a key-value pair to the operand stack
+        adv_push.8
+        # => [KEY, VALUE, remaining_entries, slot_idx, MAP_ROOT]
```

Comment:
Related to one of the previous comments: we get `[KEY, VALUE]` here because the advice map entry keeps them as `[VALUE, KEY]` but when we use `adv_push.8` instruction, the order gets naturally reversed, right?